src/transactions/test/ChangeTrustTests.cpp

```diff
@@ -277,5 +425,53 @@ TEST_CASE("change trust", "[tx][changetrust]")
             REQUIRE_THROWS_AS(root.changeTrust(idrUsd, 10),
                               ex_CHANGE_TRUST_MALFORMED);
         });
+
+        for_versions_from(18, *app, [&] {
+            // CHANGE_TRUST_MALFORMED tests
+            auto invalidFee = makeChangeTrustAssetPoolShare(
+                idr, usd, LIQUIDITY_POOL_FEE_V18 - 1);
+            REQUIRE_THROWS_AS(root.changeTrust(invalidFee, 10),
+                              ex_CHANGE_TRUST_MALFORMED);
+            auto invalidOrder =
+                makeChangeTrustAssetPoolShare(usd, idr, LIQUIDITY_POOL_FEE_V18);
+            REQUIRE_THROWS_AS(root.changeTrust(invalidOrder, 10),
+                              ex_CHANGE_TRUST_MALFORMED);
+
+            auto invalidOrderNative = makeChangeTrustAssetPoolShare(
+                idr, makeNativeAsset(), LIQUIDITY_POOL_FEE_V18);
+            REQUIRE_THROWS_AS(root.changeTrust(invalidOrderNative, 10),
+                              ex_CHANGE_TRUST_MALFORMED);
+
+            auto sameAssets =
+                makeChangeTrustAssetPoolShare(idr, idr, LIQUIDITY_POOL_FEE_V18);
+            REQUIRE_THROWS_AS(root.changeTrust(sameAssets, 10),
+                              ex_CHANGE_TRUST_MALFORMED);
+
+            // create a pool trustline with a limit of 0
+            REQUIRE_THROWS_AS(root.changeTrust(idrUsd, 0),
+                              ex_CHANGE_TRUST_INVALID_LIMIT);
+
+            SECTION("new pool with two non-native assets")
```

Comment:
should probably add some tests to cover
* when issuer is involved
* when there are multiple pools (as to test that refcounting properly works)

NB: I'm not sure which tests are slated for now vs later.
My recommendation is to start tracking tasks for missing tests in issues. I imagine that in this PR you want to have at least 1 test that tests the scenario above

Reply:
I added the tests. I also created a google doc to keep track of missing tests - https://docs.google.com/document/d/159r--a51ljU3lMgesEKdfcRYSz8ZwrQsjvBPgSOFshQ/edit. This is what we've done in the past and is easier than tracking them in github.